README.md

```diff
@@ -109,6 +109,43 @@ const catWatch = new ServiceClient({
 });
 ```
 
+## Retry Logic
+
+For application critical requests it can be a good idea to retry failed requests to the responsible services.
+
+Occasionaly target server can have high latency for a short period of time, or in the case of a stack of servers, one server can be having issues
+and retrying the request will allow perron to attempt to access one of the other servers that currently aren't facing issues.
+
+By default `perron` has retry logic implemented, but configured to perform 0 retries. Internally `perron` uses [node-retry](https://github.com/tim-kos/node-retry) to handle the retry logic
+and configuration. All of the existing options provided by `node-retry` can be passed via configuration options through `perron`.
+
+There is a shouldRetry function which can be defined in any way by the consumer and is used in the try logic to determine whether to 
+attempt the retries or not depending on the type of error. If the function returns true and the number of retries hasn't been exceeded, the request can be retried.
+
+There is also an onRetry function which can be defined by the user of `perron`. This function is called every time a retry request will be triggered.
+It is provided the currentAttempt, as well as the error that is causing the retry.
+
+```js
+const ServiceClient = require('perron');
+
+const catWatch = new ServiceClient({
+    hostname: 'catwatch.opensource.zalan.do',
+    retryOptions: {
+        retries: 1,
+        factor: 2,
+        minTimeout: 200,
+        maxTimeout: 400,
+        randomize: true,
+        shouldRetry(err) {
```

Comment:
Thinking about it, just the error might not be enough. You might need info from original request as well to make a judgement (e.g. only retry GET requests)

Reply:
Good point, I'll add the original request as well :)